js/initialize.jsx

```diff
@@ -0,0 +1,147 @@
+'use strict'
+
+var React = require('react/addons')
+var ipc = window.require('remote').require('ipc')
+
+var $ = require('jquery-bf')
+var bs = require('bootstrap')
+
+var Initform = React.createClass({
+  displayName: 'Initform',
+  mixins: [React.addons.LinkedStateMixin],
+
+  getInitialState: function () {
+    return {
+      error: null,
+      directory: '',
+      initState: 'uninitialized',
+      keysize: 2048
+    }
+  },
+
+  componentDidMount: function () {
+    var t = this
+    ipc.on('default-directory', function (dir) {
+      t.setState({directory: dir})
+    })
+
+    ipc.on('initializing', function () {
+      t.setState({initState: 'initializing'})
+    })
+
+    ipc.on('initialization-error', function (err) {
+      t.setState({
+        error: err,
+        initState: 'uninitialized'
+      })
+    })
+
+    ipc.on('initialization-complete', function () {
+      t.setState({initState: 'initialized'})
+    })
+  },
+
+  setKeySize: function (size) {
+    var t = this;
+    return function () {
+      t.setState({keysize: size})
+    }
+  },
+
+  initialize: function () {
+    ipc.emit('initialize', {
+      directory: this.state.directory,
+      keysize: this.state.keysize
+    })
+  },
+
+  render: function () {
+    var t = this
+    var initializing = (t.state.initState === 'initializing')
+    var initialized = (t.state.initState === 'initialized')
+    var uninitialized = (t.state.initState === 'uninitialized')
+
+    var description = 'Before starting, ipfs needs to generate a keypair, and create a directory to live in.'
```

Comment:
Separately, i think we want to add two checkboxes to this initial "setup dialog":

- [x] Add IPFS Node to Login Items[0]. This opens the IPFS menubar item. If it is switched to "On", it will also start your IPFS Node.[1]

[0] this is platform specific. may need different language per platform.
[1] there's got to be a better, simpler way to say this. the tricky part is it must capture (a) the on/off switch, (b) "IPFS Node" app remembering the state user left it in (on/off), (c) that it will start the IPFS node iff the node was left "On". (all without being too verbose. design is hard :) ).